wdl/gwas/regenie_summary.wdl

```diff
@@ -0,0 +1,224 @@
+
+        def get_gnomad_annotation(iterator,cpra,finaf_idx, nfeaf_idx, rsid_idx,gd_cpra) -> GnomadAnnotation:
+            for v in iterator:
+                cols = v.strip("\n").split("\t")
+
+                if (cols[gd_cpra[0]] == "chr"+cpra[0]) and (cols[gd_cpra[1]] == str(cpra[1])) and (cols[gd_cpra[2]] == cpra[2]) and (cols[gd_cpra[3]] == cpra[3]):
+                    return GnomadAnnotation(cols[finaf_idx],cols[nfeaf_idx],cols[enrich_idx],cols[rsid_idx])
+            return GnomadAnnotation(".",".",".",".")
+
+        #required columns
+        fg_req_cols=["#variant","gene_most_severe","most_severe"]
+        gd_req_cols=["fin.AF","nfsee.AF","enrichment_nfsee","rsid"]
+        #open finngen annotation tabix
+        fg_tabix = pysam.TabixFile(finngen_annotation_file,parser=None)
+        #get fg header column positions
+        fg_header = get_header(gzip.open, finngen_annotation_file)
+        fg_idx = {v:i for (i,v) in enumerate(fg_header)}
+        #open gnomad annotation tabix
+        gnomad_tabix = pysam.TabixFile(gnomad_annotation_file,parser=None)
+        gd_header = get_header(gzip.open, gnomad_annotation_file)
+        gd_idx = {v:i for (i,v) in enumerate(gd_header)}
+        gd_cpra = (
+            gd_idx["chrom"],
+            gd_idx["pos"],
+            gd_idx["ref"],
+            gd_idx["alt"]
+        )
+        #check for fg column existence
+        if not all([a in fg_header for a in fg_req_cols]):
+            raise Exception("Not all columns present in FinnGen annotation! Aborting...")
+        #check for gnomad column existence
+        if not all([a in gd_header for a in gd_req_cols]):
+            raise Exception("Not all columns present in Gnomad annotation! Aborting...")
+        var_idx, gene_idx, cons_idx = (fg_idx["#variant"],fg_idx["gene_most_severe"],fg_idx["most_severe"])
+
+        finaf_idx, nfeaf_idx, enrich_idx, rsid_idx = (gd_idx["fin.AF"],gd_idx["nfsee.AF"],gd_idx["enrichment_nfsee"],gd_idx["rsid"])
+
+
+
+        with gzip.open(fname, "rt") as file:
+            #open output file
+            with open(output_summary_name,"w") as summary_outfile, open(output_coding_name,"w") as coding_outfile:
+                #read header
+                header = file.readline().strip("\n").split('\t')
+                #find p-value index
+                header_idx = {v:i for (i,v) in enumerate(header)}
+                pval_idx = header_idx["pval"]
+                cid,pid,rid,aid = (
+                    header_idx["#chrom"],
+                    header_idx["pos"],
+                    header_idx["ref"],
+                    header_idx["alt"]
+                )
+
+                #add gene name, consequence, gnomad finnish af, nfsee af, rsid
+                header.extend(["fin.AF","nfsee.AF","rsid","gene_most_severe","most_severe"])
+                summary_outfile.write("\t".join(header)+"\n")
+                #add pheno, fin enrichment
+                header.extend(["fin.enrichment","phenotype"])
```

Comment:
The last 2 columns are merged iin the provided results, but looks fine here
gsutil cat gs://fg-cromwell_fresh/regenie_summaries/29e318d8-92dd-435c-9c41-4c17f200157d/call-post_summary/summary.regenie_summary/164d4497-6b05-4055-b0af-e704f2b4b353/call-gather/coding_variants.txt.gz | zcat | head

Reply:
I don't know what you mean, seems fine to me:
![image](https://user-images.githubusercontent.com/10233157/144370980-689768de-1da1-4edb-9541-0976653addf4.png)

Reply:
wtf..... dunno how on earth they were shown together... must be some terminal quirkiness. anyhoo